### ISSUE TITLE
Update to processors 8.5.0 will break two Eidos unit tests

### ISSUE THREAD

**Issue by kwalcock** [clulab/processors#621]
The new version will not find mentions in the sentence in TestRaps:
```
Significant decline in poverty will be associated with a decrease in family size and increase in non-farm income.
```

for
```
    Concept, Entity => family size
     * Attachments: Decrease(decrease,None)
```
and
```
    Concept, Entity => non-farm income
     * Attachments: Increase(increase,None)
```
If that's no big deal, I will just disable the test.  If it's worth looking into, here are some details.  PDFs of the entire pages are attached.  Here are just the top portions from before

![image](https://user-images.githubusercontent.com/8679738/165446118-1e663252-bf12-4f49-bb47-1d604f6e391b.png)

and after

![image](https://user-images.githubusercontent.com/8679738/165446231-2724e09e-f18a-452d-bb94-07ddb42774e4.png)

This is jumping from processors 8.4.7 to 8.5.0, so the change might not have happened with the very most recent release.

[World Modelers Visualizer-after.pdf](https://github.com/clulab/processors/files/8569045/World.Modelers.Visualizer-after.pdf)
[World Modelers Visualizer-before.pdf](https://github.com/clulab/processors/files/8569046/World.Modelers.Visualizer-before.pdf)

**Comment by kwalcock** [comment]
The only other test that fails is less specific.  The TestSentenceClassifier complains that
```
precision and recall: 0.7941176, 0.18181819

0.7941176 was not greater than 0.8
```
It was previously at
```
precision and recall: 0.80597013, 0.18181819
```

**Comment by MihaiSurdeanu** [comment]
I think it's fine to adjust the SentenceClassifier test to allow P > 0.79.
More on the other soon.

**Comment by kwalcock** [comment]
OK.  That other is not a critical test or anything, but it might be indicative of some unforeseen behavior/feature.  I thought it deserved a check before anything was published.

**Comment by MihaiSurdeanu** [comment]
Also,
- Another student reported a drop in scores when switching to a newer processors/fatdynet. I suspect changes in dynet impacted things. I asked her for details. However, I do not observe this when training the processors models, so I think we can ignore it until we have more evidence.
- On the second issue: the output was clearly better before. I am not sure why... The only difference is that the amod from "income" to "non-farm" is lost in the new output, and it's replaced by another incorrect amod from "income" to "Significant". I am not sure this is sufficient for the change... I am not sure why this difference exists, since we use the same version of CoreNLP. Any ideas?

**Comment by kwalcock** [comment]
It looks like the parse is different before and after (in the PDFs).  Before is non-farm / income and after is non / farm / income.  It looks like brat isn't drawing the diagram correctly because it can't find "non" or "farm" (It may only see "non-farm") and uses 0 or -1 for an index to draw the line all the way back to "Significant".  That line is not listed in the dependencies, if I understand it correctly.  The hyphen must be causing a problem.

**Comment by MihaiSurdeanu** [comment]
Hmm... Can you please check where we lose the dash? This sounds like a tokenization issue. This looks vaguely familiar. Quite probably my fault.

**Comment by kwalcock** [comment]
It happens in class TokenizerStepHyphens where "non-" was added recently.

**Comment by MihaiSurdeanu** [comment]
I think I did that to help Eidos (or Reach?) handle some negations better...
Just to be extra sure, if you remove "non-" from "non-farm" from the original text, do we extract 2 causal relations?

**Comment by kwalcock** [comment]
Yes, but it's still acting strangely.  "in farm income", "in non- farm income", and "in non farm income" all result in 2 relations.  It is only when they are written together "in non-farm income" that there is a problem and the tokenizer seems to convert "non-farm" to "non farm" anyway.  I haven't yet found out what the difference is, but must be pretty close.

[World Modelers Visualizer-after-justFarm.pdf](https://github.com/clulab/processors/files/8578961/World.Modelers.Visualizer-after-justFarm.pdf)

**Comment by MihaiSurdeanu** [comment]
Hmm... I expected the output for "non-farm income" and "non farm income" to be identical because the former is tokenized into the latter.

**Comment by kwalcock** [comment]
It seems that the beginPosition and endPosition of the RawToken  for the two halves, "non" and "farm" are incorrect.  They are positions within the word and not the sentence or document, and that might be messing things up.

**Comment by kwalcock** [comment]
In the code
```scala
          tokens += RawToken(input.raw.substring(0, dashPosition), 0, dashPosition)
          tokens += RawToken(input.raw.substring(dashPosition + 1), dashPosition + 1)
```
probably the input.beginPosition needs to be added to the 0 and dashPosition.  I can change that in processors and see if it makes a difference.

**Comment by MihaiSurdeanu** [comment]
Nice catch!!

**Comment by kwalcock** [comment]
That seems to have fixed it.  The arrows are now in the right place as well.  It could easily have been a false alarm or rabbit hole, so thanks for supporting further investigation.  PRs are being prepared.

![image](https://user-images.githubusercontent.com/8679738/165673086-5d816114-eff1-4b87-ad0b-64fa149faa33.png)
![image](https://user-images.githubusercontent.com/8679738/165673376-8d6ac8d0-d5ba-424a-ad28-8a8ff9afdd9d.png)

**Comment by MihaiSurdeanu** [comment]
Nice!

**Comment by kwalcock** [comment]
I think this was fixed around #631.
